common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -112,6 +116,7 @@ public TransportClientFactory(
     }
     this.metrics = new NettyMemoryMetrics(
       this.pooledAllocator, conf.getModuleName() + "-client", conf);
+    fastFailTimeWindow = conf.maxIORetries() > 0 ? (int)(conf.ioRetryWaitTimeMs() * 0.95) : 0;
```

Comment:
so I think you may have misunderstood my comment. my concern is really on those things that don't go through RetryFetcher and don't use maxIORetries.  Like  just sending RPC messages.
For instance the external block client fetching the local host dirs, or just any other rpc message.   I guess the rpc messages through outbox all cache the client, so that isn't as much of a concern.  Looking some more, it looks like those cases are pretty limited so it should be ok.
Also this doesn't really disable it if the max retries is 0 because you could still theoretically hit the case that 2 try in the same millisecond and then the second would fail fast.  how about setting to -1 in that case

Reply:
thanks, make sense.